staging/src/k8s.io/client-go/tools/remotecommand/remotecommand_test.go

```diff
@@ -131,12 +145,31 @@ func TestSPDYExecutorStream(t *testing.T) {
 			expectError: "",
 			attacher:    fakeMassiveDataAttacher,
 		},
+		{
+			timeout: 500 * time.Millisecond,
+			name:    "timeoutTest",
+			options: StreamOptions{
+				Stdin:  &fakeMassiveDataPty{},
+				Stderr: &fakeMassiveDataPty{},
+			},
+			expectError: context.DeadlineExceeded.Error(),
+			attacher:    fakeMassiveDataAttacher,
+		},
 	}
 
 	for _, test := range tests {
 		server := newTestHTTPServer(test.attacher, &test.options)
 
-		err := attach2Server(server.URL, test.options)
+		ctx, cancelFn := context.Background(), func() {}
+		if test.timeout > 0 {
+			ctx, cancelFn = context.WithTimeout(ctx, test.timeout)
+		}
+
+		err := func(ctx context.Context, cancel context.CancelFunc) error {
+			defer cancelFn()
+			return attach2Server(ctx, server.URL, test.options)
+		}(ctx, cancelFn)
```

Comment:
Why not just

```suggestion
		err := attach2Server(ctx, server.URL, test.options)
		cancelFn()
```
?

Reply:
They are the same but i wanna defer the cancellation as package `context` recommends. 

And I have a typo 🙃. The `cancelFn` should be `cancel`.

Reply:
have you fixed the typo you mentioned?

Reply:
🥲 Sorry, I forgot to fix this. Just fixed. Now it's good, I think.